dhis-2/dhis-services/dhis-service-field-filtering/src/main/java/org/hisp/dhis/fieldfiltering/FieldFilterService.java

```diff
@@ -251,10 +276,45 @@ private <T> void toObjectNodes(
       applyAttributeAsPropertyFields(object, objectNode, paths);
       applyTransformers(objectNode, null, "", fieldTransformers);
 
+      if (excludeDefaults) removeEmptyObjects(objectNode);
+
       consumer.accept(objectNode);
     }
   }
 
+  /**
+   * Method that removes empty objects from an ObjectNode, at root level.
+   *
+   * <p>e.g. json input: <br>
+   * <code>
+   *   {
+   *      "name": "dhis2",
+   *      "system": {}
+   *   }
+   * </code>
+   *
+   * <p><br>
+   *
+   * <p>resulting output: <br>
+   * <code>
+   *   {
+   *      "name": "dhis2"
+   *   }
+   * </code>
+   *
+   * @param objectNode object node to process on
+   */
+  private void removeEmptyObjects(ObjectNode objectNode) {
```

Comment:
For this, maybe we can turn on [JsonInclude.Include.NON_EMPTY](https://github.com/dhis2/dhis2-core/blob/3f202cc74b9ceb72b17beb257fc7fb3d34bd55fa/dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/jackson/config/JacksonObjectMapperConfig.java#L165). 
Can discuss later.

Reply:
I actually tried using that in several places but it didn't work for some reason.
Used it on the mapper `.setSerializationInclusion(JsonInclude.Include.NON_EMPTY)`
and also tried at the class & field level annotation. Strange.